tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -4077,63 +4150,42 @@ std::optional<Value> convertGatherOp(PatternRewriter& rewriter, Operation* op,
                                            params_type.getElementType()),
       params_value, params_transpose_perm_val.value());
 
-  if (count_dynamic_dims(tosa_values_shape) > 1) {
-    return (void)rewriter.notifyMatchFailure(
-               op,
-               "multiply dynamic shapes when reshaping values down to "
-               "tosa.gather"),
-           std::nullopt;
+  std::optional<Value> tosa_values_reshape_op = build_reshape(
+      rewriter, op->getLoc(), params_transpose_op, {N, K, C}, tosaOnly);
+
+  if (!tosa_values_reshape_op) {
```

Comment:
Lets make `build_reshape` always return a value that way we do not have to check-bail each construction.